src/lib.rs

```diff
@@ -519,6 +521,11 @@ mod contract {
         EthConnectorContract::get_instance().get_accounts_counter()
     }
 
+    #[no_mangle]
+    pub extern "C" fn migrate() {
+        crate::migration::migrate()
+    }
+
```

Comment:
Do all I/O-related here, please.

```suggestion
    pub extern "C" fn migrate() {
        sdk::assert_private_call();
        let args = Migration::from(parse_json(&sdk::read_input()).expect_utf8(ERR_FAILED_PARSE.as_bytes()));
        crate::migration::migrate(args);
        sdk::return_output(&"done".as_bytes());
    }
    
```